fla/utils.py

```
@@ -82,9 +84,66 @@ def wrapper(*args, **kwargs):
+    return triton.runtime.driver.active.utils.get_device_properties(tensor_idx)['multiprocessor_count']
+
+
+@lru_cache(maxsize=None)
+def get_available_device() -> str:
+    if torch.cuda.is_available():
+        return "cuda"
+
+    try:
+        import intel_extension_for_pytorch as ipex  # noqa: F401
+    except ImportError:
+        pass
```

Ruff (0.8.2)

Use `contextlib.suppress(ImportError)` instead of `try`-`except`-`pass`

(SIM105)